docs/users/topics/phpstorm.md

```diff
@@ -0,0 +1,50 @@
+## PhpStorm Integration
+
+This topic explores how to add full PhpStorm integration with a DDEV project, including composer and phpunit.
+
+This is based on the wonderful [article](https://susi.dev/fully-integrate-ddev-and-phpstorm-including-unit-tests-with-coverage-update-2021) by Susi, and couldn't have been done without it, but since PhpStorm has fixed some things and DDEV has worked around others, this is far easier now. It works on macOS, Linux, Windows with WSL2 and Windows traditional. You'll end up with PhpStorm actually using the PHP interpreter inside the ddev-webserver container of your project, and able to use composer and phpunit inside there as well.
+
+### Requirements
+
+- PhpStorm 2021.2 or higher
+- docker-compose v1. Due to a [bug or two in PhpStorm](https://youtrack.jetbrains.com/issue/WI-61205), it can't properly interpret docker-compose v2 output as of Sept 2021.  Use `docker-compose disable-v2` to switch to docker-compose v1. (Check version with `docker-compose --version`.)
+- Any OS platform, but if you're using Windows PhpStorm with WSL2 the path mappings are slightly more complex. WSL2 instructions are provided where necessary.
+
+Steps:
+
+1. Open a DDEV project. In this example, the project name is "d9" and the site is "d9.ddev.site".
+    - If you're on Windows, running PhpStorm on the Windows side but  using WSL2 for your DDEV project, open the project as a WSL2 project. In other words, in the "Open" dialog, browse to `\\wsl$\Ubuntu\home\rfay\workspace\d9` (in this example). (If you're running PhpStorm inside WSL2, there are no special instructions.)
+2. Set up your project to do normal Xdebug, as described in the [Step Debugging section](../step_debugging.md). This will result in a PhpStorm "Server" with the proper name, normally the same as the FQDN of the project. In this example, "d9.ddev.site". (All you have to do here is click the little telephone to "Start listening for PHP Debug Connections", then `ddev xdebug on`, then visit a web page and choose the correct mapping from host to server. )
+3. Under File→Settings→PHP (Windows) or Preferences→PHP (macOS), click the "..." to the right of "CLI Interpreter"
+    1. Use the "+" to select "From Docker, Vagrant, VM..."
+    2. Choose "Docker Compose"
+    3. Create a server; the default name is "docker", but since the "server" for each project will be different, name it for the project, for example "DDEV d9". Choose "Docker for Windows" or "Docker for Mac"
+    4. In the "Path mappings" of the "Server" map the local paths (which on WSL2 means /home/...) to the in-container paths. So "Virtual Machine Path" would be "/var/www/html" and "Local path" would be something like /Users/rfay/workspace/d9 (on macOS) or \\wsl$\Ubuntu\home\rfay\workspace\d9 on Windows using WSL2
```

Comment:
When you set the compose file in step 5, aren't the path mappings set up automatically? So far they have been for us on macOS - we won't be trying things out on the Windows side for a while, if ever.

Reply:
Oh, I was probably testing with mutagen enabled, which would prevent PhpStorm from being able to derive the main mount.

Reply:
```suggestion
    4. In the "Path mappings" of the "Server" you may have to map the local paths (which on WSL2 means /home/...) to the in-container paths, especially if you have mutagen enabled. So "Virtual Machine Path" would be "/var/www/html" and "Local path" would be something like /Users/rfay/workspace/d9 (on macOS) or \\wsl$\Ubuntu\home\rfay\workspace\d9 on Windows using WSL2.
```

Reply:
That would make sense! Also, I think path mappings will be preserved if you switch from a normal mount to using mutagen on an existing project.